orchagent/portsorch.cpp

```diff
@@ -1452,7 +1452,7 @@ bool PortsOrch::bake()
     if (portCount != keys.size() - 2)
     {
         // Invalid port table
-        SWSS_LOG_ERROR("Invalid port table: portCount, expecting %" PRIu64 ", got %" PRIu64,
+        SWSS_LOG_ERROR("Invalid port table: portCount, expecting %" PRIu64 ", got %zu",
```

Comment:
>PRIu64 [](start = 68, length = 6)

%lu is the correct format for unsigned long. Then you don't need change type of portCount

Reply:
portCount is declared as size_t and not as ulong.

Reply:
%zu is for (keys.size()-2) which is of type size_t and hence %zu has to be used.

Reply:
I am talking about `portCount` format string `PRIu64` -> '%lu', and no need to change its original type `unsigned long`

---
In reply to: [309990956](https://github.com/Azure/sonic-swss/pull/1015#discussion_r309990956) [](ancestors = 309990956)

Reply:
I have updated the code by changing stoul with strtoumax() API to avoid using ulong.